### PR TITLE
Bubble up MachineDeployment ready condition and print current/desirednode

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This improve print columns to show current/desired nodes.
Also it bubbles up the ready condition from MachineDeployments which should aggregate state from the in progress operations and underlying MachineSet and Machines.
As a consequence it move from metav1.condition into our own struct definition since metav1.Condition has validation for Reason that might be broken by what we bubble up from CAPI.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
e.g

```
- lastTransitionTime: "2022-03-03T18:07:54Z"
    message: Minimum availability requires 8 replicas, current 7 available
    observedGeneration: 4
    reason: WaitingForAvailableMachines
    status: "False"
    type: Ready
  nodeCount: 7
  version: 4.10.0
```

File: hypershift-operator/main.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		Scheme:                        hyperapi.Scheme,
 		MetricsBindAddress:            opts.MetricsAddr,
 		Port:                          9443,
-		LeaderElection:                true,
+		LeaderElection:                false,
```
Comment (ironcladlou) [inline hypershift-operator/main.go:147]: Looks like maybe an unintentional debug commit? I'd love to have a flag for this btw

File: hypershift-operator/main.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		Scheme:                        hyperapi.Scheme,
 		MetricsBindAddress:            opts.MetricsAddr,
 		Port:                          9443,
-		LeaderElection:                true,
```
Comment (alvaroaleman) [inline hypershift-operator/main.go:147]: Is this intended? Other than that, the change lgtm

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 3]
```diff
@@ -812,7 +812,30 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 
 	setMachineDeploymentReplicas(nodePool, machineDeployment)
 
+	// Bubble up AvailableReplicas and Ready condition from MachineDeployment.
```
Comment (csrwng) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:815]: @enxebre this looks fine to me, but it does diverge from the pattern of reconciling all status at the beginning of the reconcile function. If that's ok, then lgtm

Comment (enxebre) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:815]: Good point, should be fine for now, we can then consider move it up along with MHC condition.

The underlying condition contains invalid characters for metav1.Condition.Reason so I'm now looking into this, possibly need to create our own conditions type.

Comment (enxebre) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:814]: Updated to move into our own condition struct.
In a follow up I will consider conflate some of the existing conditions and move dep/mhc up, drawback would be additional api server overhead.

**Pull request merged**